app/models/CreateGoodsRecord.scala

```diff
@@ -49,14 +64,22 @@ object GoodsRecord {
       case Left(errors)      => Left(errors)
     }
 
-  private def getCommodityCode(answers: UserAnswers): EitherNec[ValidationError, String] =
+  private def getCommodity(answers: UserAnswers): EitherNec[ValidationError, Commodity] =
     answers.getPageValue(CommodityCodePage) match {
-      case Right(data)  =>
+      case Right(code)  =>
         answers.getPageValue(HasCorrectGoodsPage) match {
-          case Right(true)  => Right(data)
+          case Right(true)  => getCommodityQuery(answers, code)
```

Comment:
Ah I thought the commodity page was returning a string rather than commodity

Reply:
Have a read through how the commodity object is gotten - you will see that is uses the page and the query and then checks they both have the same code